File: prodplus-api/src/main/java/br/com/prodplus/models/Configuracao.java
package br.com.prodplus.models;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Id;

import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import com.fasterxml.jackson.annotation.JsonIgnore;

import br.com.prodplus.models.auxiliares.Turno;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 
 * @author Marlon F. Garcia
 *
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Configuracao implements Serializable {

	private static final long serialVersionUID = -4172693040286698727L;
	@Id
	@JsonIgnore
	private Integer id;
	@ElementCollection
	@LazyCollection(LazyCollectionOption.FALSE)
	private Set<Turno> turnosSemana = new HashSet<>();
	@ElementCollection
	@LazyCollection(LazyCollectionOption.FALSE)
	private Set<Turno> turnosSabado = new HashSet<>();
	@ElementCollection
	@LazyCollection(LazyCollectionOption.FALSE)
	private Set<Turno> turnosDomingo = new HashSet<>();
	private boolean estruturaCompleta = false;
	private Double mediaIpcm = 0.0;

}

File: prodplus-front/src/app/models/configuracao.ts
import { Turno } from "./auxiliares/turno";

export class Configuracao {
  turnosSemana: Turno[];
  turnosSabado: Turno[];
  turnosDomingo: Turno[];
  estruturaCompleta: boolean;
  mediaIpcm: number;

  constructor(turnosSemana: Turno[], turnosSabado: Turno[], turnosDomingo: Turno[],
      estruturaCompleta: boolean, mediaIpcm: number) {
    this.turnosSemana = turnosSemana;
    this.turnosSabado = turnosSabado;
    this.turnosDomingo = turnosDomingo;
    this.estruturaCompleta = estruturaCompleta;
    this.mediaIpcm = mediaIpcm;
  }
}

export function ordenaTurnos(turnos: Turno[]): Turno[] {
  turnos.sort((a, b) => a.inicio > b.inicio ? 1 : a.inicio < b.inicio ? -1 : 0);
  return turnos;
}
